addons/payment_alipay/models/payment_acquirer.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from hashlib import md5
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(selection_add=[
+        ('alipay', 'Alipay')
+    ], ondelete={'alipay': 'set default'})
+    alipay_payment_method = fields.Selection([
+        ('express_checkout', 'Express Checkout (only for Chinese Merchant)'),
+        ('standard_checkout', 'Cross-border'),
+    ], string='Account', default='express_checkout',
+        help="* Cross-border: For the Overseas seller \n* Express Checkout: For the Chinese Seller")
+    alipay_merchant_partner_id = fields.Char(
+        string='Merchant Partner ID', required_if_provider='alipay', groups='base.group_system',
+        help='Alipay account found in the merchant portal')
+    alipay_md5_signature_key = fields.Char(
+        string='MD5 Signature Key', required_if_provider='alipay', groups='base.group_system',
+        help="A 32-byte ascii string used to sign the communications with Alipay.")
+    alipay_seller_email = fields.Char(string='Alipay Seller Email', groups='base.group_system')
+
+    @api.model
+    def _get_alipay_urls(self):
+        """ Alipay URLS """
+        if self.state == 'enabled':
+            return 'https://mapi.alipay.com/gateway.do'
+        else:
+            # Testing environment
+            return 'https://openapi.alipaydev.com/gateway.do'
+
+    def _compute_fees(self, amount, currency_id, country_id):
```

Comment:
This method will get a new signature today:
```suggestion
    def _compute_fees(self, amount, currency, country):
```
With `currency` and `country` non-None records (possibly empty recordsets)